decidim-proposals/app/controllers/concerns/decidim/proposals/orderable.rb

```diff
@@ -28,17 +28,18 @@ def available_orders
         end
 
         def default_order
-          if current_settings.votes_blocked?
-            detect_order("most_voted")
-          else
-            "random"
-          end
+          return detect_order("most_voted") if votes_blocked?
+          "random"
         end
 
         def votes_visible?
           current_settings.votes_enabled? && !current_settings.votes_hidden?
         end
 
+        def votes_blocked?
+          votes_visible? && current_settings.votes_blocked?
+        end
```

Comment:
I think the name of these utility methods is misleading, because it uses very similar (or exactly the same) terminology for different concepts. `votes_visible?` should be called `most_voted_order_available?` and `votes_blocked?` should be called `order_by_votes?`. Or something.

But the fix is good indeed!